freqtrade/rpc/rpc.py

```diff
@@ -1130,15 +1131,15 @@ def _rpc_list_custom_data(
         if not trades:
             return []
 
-        # Collect custom data
         custom_data = []
-        if key:
-            data = trades.get_custom_data(key=key)
-            if data:
-                custom_data = [data]
-        else:
-            for trade in trades:
-                custom_data.extend(trade.get_all_custom_data())
+        for trade in trades:
+            # Collect custom data
+            if key:
+                data = trade.get_custom_data(key=key)
+                if data:
+                    custom_data = [data]
+
+            custom_data.extend(trade.get_all_custom_data())
```

Comment:
@Axel-CH i think you'll want to keep this in the else - this way you're ignoring the "key" premise by always querying for "everything" and extending `custom_data` with it.